enterprise/ssl-impl/src/test/java/io/crate/protocols/mqtt/CrateMqttTransportIntegrationTest.java

```diff
@@ -0,0 +1,135 @@
+    public static void afterIntegrationTest() {
+        System.clearProperty("javax.net.ssl.trustStore");
+        System.clearProperty("javax.net.ssl.trustStorePassword");
+    }
+
+    @Override
+    protected Settings nodeSettings(int nodeOrdinal) {
+        return Settings.builder()
+            .put(super.nodeSettings(nodeOrdinal))
+            .put(SharedSettings.ENTERPRISE_LICENSE_SETTING.getKey(), true)
+            .put(Netty4MqttServerTransport.MQTT_ENABLED_SETTING.getKey(), true)
+            .put(Netty4MqttServerTransport.MQTT_PORT_SETTING.getKey(), 8883)
+            .put(SslConfigSettings.SSL_MQTT_ENABLED.getKey(), true)
+            .put(SslConfigSettings.SSL_KEYSTORE_FILEPATH.getKey(), keyStoreFile)
+            .put(SslConfigSettings.SSL_KEYSTORE_PASSWORD.getKey(), "keystorePassword")
+            .put(SslConfigSettings.SSL_KEYSTORE_KEY_PASSWORD.getKey(), "serverKeyPassword")
+            .put(SslConfigSettings.SSL_TRUSTSTORE_FILEPATH.getKey(), trustStoreFile)
+            .put(SslConfigSettings.SSL_TRUSTSTORE_PASSWORD.getKey(), "truststorePassword")
+            .build();
+    }
+
+    @Before
+    public void setupMqttTableAndIngestRule() {
+        execute("CREATE TABLE IF NOT EXISTS mqtt.raw (\n" +
+                "  client_id STRING,\n" +
+                "  packet_id INTEGER,\n" +
+                "  topic STRING,\n" +
+                "  ts TIMESTAMP,\n" +
+                "  payload OBJECT(DYNAMIC),\n" +
+                "  PRIMARY KEY (\"client_id\", \"packet_id\")\n" +
+                ") WITH (column_policy = 'strict', number_of_replicas = '0')");
+
+        execute("CREATE INGEST RULE ingest_topic_t1 ON " + MqttIngestService.SOURCE_IDENT +
+                " WHERE topic = ? INTO mqtt.raw", new Object[]{"t1"});
+    }
+
+    @Before
+    public void setupMqttClient() throws MqttException {
+        client = new MqttClient("ssl://localhost:8883", "sslTestClient", null);
+        client.setTimeToWait(5000);
+        MqttConnectOptions options = new MqttConnectOptions();
+        options.setConnectionTimeout(60);
+        options.setKeepAliveInterval(60);
+        options.setSocketFactory(SSLSocketFactory.getDefault());
+        client.connectWithResult(options).waitForCompletion();
+    }
+
+    @After
+    public void dropIngestRule() {
+        execute("DROP INGEST RULE IF EXISTS ingest_topic_t1");
+    }
+
+    @After
+    public void disconnectAndCloseMqttClient() throws MqttException {
+        client.disconnect();
+        client.close();
+    }
+
+    @Test
+    public void testPublishEncryptedMessage() throws Exception {
+        client.publish("t1", "{\"id\": 123}".getBytes(), 1, false);
+
+        assertBusy(() -> {
+            // busy asserting as there's now way to wait for puback with this mqtt client
+            execute("select payload['id'] from mqtt.raw");
+            assertThat(response.rowCount(), is(1L));
+            assertThat(response.rows()[0][0], is(123L));
+        }, 3, TimeUnit.SECONDS);
```

Comment:
This is bound to be a flaky test :)

Reply:
Wouldn't it be possible to signal back from the MQTT client? We don't have to test inserts into Crate. We just want to confirm message receipt.

Reply:
Ok, so this and the previous comment are due to the fact that the mqtt client doesn't offer any way to wait for a PUBACK or reply for a publish (as I've indicated in the comment 4 lines above :) ) 
I'll have another look at the client, maybe it does and I've missed it